src/libbson/src/bson/bson-vector.h

```diff
@@ -0,0 +1,591 @@
+   if (BSON_LIKELY (vector_offset_bytes <= length_bytes && byte_count <= length_bytes - vector_offset_bytes)) {
+      if (byte_count == length_bytes - vector_offset_bytes && byte_count >= 1u) {
+         // This write touches the last byte in the vector:
+         // special-case that byte so we can ensure unused bits remain set to zero.
+         size_t other_bytes = byte_count - 1u;
+         memcpy (view.binary.data + BSON_VECTOR_HEADER_LEN + vector_offset_bytes, packed_values, other_bytes);
+         view.binary.data[BSON_VECTOR_HEADER_LEN + vector_offset_bytes + other_bytes] =
+            ((uint8_t) 0xFF << bson_vector_packed_bits_view_padding (view)) & packed_values[other_bytes];
+      } else {
+         memcpy (view.binary.data + BSON_VECTOR_HEADER_LEN + vector_offset_bytes, packed_values, byte_count);
+      }
+      return true;
+   } else {
+      return false;
+   }
+}
+
+
+static BSON_INLINE bool
+bson_vector_packed_bits_const_view_unpack_bool (bson_vector_packed_bits_const_view_t view,
+                                                bool *BSON_RESTRICT unpacked_values_out,
+                                                size_t element_count,
+                                                size_t vector_offset_elements)
+{
+   size_t length = bson_vector_packed_bits_const_view_length (view);
+   if (BSON_LIKELY (vector_offset_elements <= length && element_count <= length - vector_offset_elements)) {
+      size_t i;
+      for (i = 0; i < element_count; i++) {
+         size_t element_index = vector_offset_elements + i;
+         uint8_t packed_byte = view.binary.data[BSON_VECTOR_HEADER_LEN + (element_index >> 3)];
+         unpacked_values_out[i] = 0 != (packed_byte & ((uint8_t) 0x80 >> (element_index & 7)));
+      }
+      return true;
+   } else {
+      return false;
+   }
+}
+
+static BSON_INLINE bool
+bson_vector_packed_bits_view_unpack_bool (bson_vector_packed_bits_view_t view,
+                                          bool *BSON_RESTRICT unpacked_values_out,
+                                          size_t element_count,
+                                          size_t vector_offset_elements)
+{
+   return bson_vector_packed_bits_const_view_unpack_bool (
+      bson_vector_packed_bits_view_as_const (view), unpacked_values_out, element_count, vector_offset_elements);
+}
+
+static BSON_INLINE bool
+bson_vector_packed_bits_view_pack_bool (bson_vector_packed_bits_view_t view,
+                                        const bool *BSON_RESTRICT unpacked_values,
+                                        size_t element_count,
+                                        size_t vector_offset_elements)
+{
+   size_t length = bson_vector_packed_bits_view_length (view);
+   if (BSON_LIKELY (vector_offset_elements <= length && element_count <= length - vector_offset_elements)) {
+      while (element_count > 0) {
+         uint8_t *BSON_RESTRICT packed_byte = &view.binary.data[BSON_VECTOR_HEADER_LEN + (vector_offset_elements >> 3)];
```

Comment:
```suggestion
         uint8_t *BSON_RESTRICT packed_byte = view.binary.data + BSON_VECTOR_HEADER_LEN + (vector_offset_elements >> 3);
```
Minor suggestion to avoid an intermediate subscript.